pkg/apis/awsnodetemplate/v1alpha1/awsnodetemplate.go

```diff
@@ -29,9 +29,15 @@ type AWSNodeTemplateSpec struct {
 	// +optional
 	UserData     *string `json:"userData,omitempty"`
 	v1alpha1.AWS `json:",inline"`
-	// AMISelector discovers AMIs to be used by Amazon EC2 tags.
+	// AMISelector discovers AMIs to be used by filters.
 	// +optional
-	AMISelector map[string]string `json:"amiSelector,omitempty"`
+	AMISelector *AMISelector `json:"amiSelector,omitempty"`
+}
+
+// AMISelector discovers AMIs to be used by filters
+type AMISelector struct {
+	AMIFilters map[string]string `json:"amiFilters,omitempty"`
+	MostRecent *bool             `json:"mostRecent,omitempty"`
```

Comment:
This is a backwards incompatible change, and will require a v1alpha6 bump. See: #1327. One of our tenets in https://karpenter.sh/docs/contributing/design-guide/ is that we should only expose parameters/functionality as necessary. Does it make sense to just default to most recent? This seems like a reasonable experience to me, and doesn't require API changes.

Reply:
Okay. Yes I think it should default to the most recently created AMI. I'll push up another change to set `AMISelector` back to `map[string]string` with the logic to use the most recently created AMI.